vendor/processors/stripe_processor.py

```diff
@@ -341,6 +349,10 @@ def build_price(self, offer, price):
         if 'stripe' not in offer.meta or 'product_id' not in offer.meta['stripe']:
             raise TypeError(f"Price cannot be created without a product_id on offer.meta['stripe'] field")
 
+        if isinstance(price, (int, float)):
+            raise TypeError(f"Price cannot be created without a valid vendor price object added to this offer")
+
+
         price_data = {
             'product': offer.meta['stripe']['product_id'],
             'currency': price.currency,
```

Comment:
If you do not have a price instance because the offer does not have one you can use this value for the currency
`from vendor.config import DEFAULT_CURRENCY`